packages/ui/src/admin/ModelForm/ModelCreateForm.tsx

```diff
@@ -115,7 +115,7 @@ export const ModelCreateForm = <T extends StrapiModel>({
     const bodyData = {
       ...body,
       slug,
-      publishedAt: url !== 'api/recommended-topics' ? null : new Date(),
+      publishedAt: url !== ('api/recommended-topics' ) ? url !=='api/recommended-tweets'?null:new Date() : new Date(),
```

Comment:
Wouldn't it be better to control publish state via props?. We can pass `shouldPublish` to `ModelCreateModal` and `ModalCreateForm`

```suggestion
      publishedAt: shouldPublish ? new Date() : null,
```